Index: src/main/java/com/maple/study/pattern/chain/Women.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/Women.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/Women.java	(revision )
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public class Women implements IWomen {
+
+    /*
+    1: 未嫁
+    2: 已嫁
+    3: 丧偶
+     */
+    private int type = 0;
+    private String request = "";
+
+    public Women(int type, String request) {
+        this.type = type;
+        this.request = request;
+        switch (type) {
+            case 1:
+                this.request = "女儿的请求是: " + request;
+                break;
+            case 2:
+                this.request = "妻子的请求是: " + request;
+                break;
+            case 3:
+                this.request = "母亲的请求是: " + request;
+                break;
+        }
+    }
+
+    @Override
+    public int getType() {
+        return this.type;
+    }
+
+    @Override
+    public String getRequest() {
+        return this.request;
+    }
+}
Index: src/main/java/com/maple/study/pattern/chain/Handler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/Handler.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/Handler.java	(revision )
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public abstract class Handler {
+    public final static int FATHER_LEVEL_REQUEST = 1;
+    public final static int HUSBAND_LEVEL_REQUEST = 2;
+    public final static int SON_LEVEL_REQUEST = 3;
+
+    private int level = 0;
+
+    private Handler nextHandler;
+
+    public Handler(int level) {
+        this.level = level;
+    }
+
+    public final void handleMessage(IWomen women) {
+        if (women.getType() == this.level) {
+            this.response(women);
+        } else {
+            if (this.nextHandler != null) {
+                this.nextHandler.handleMessage(women);
+            } else {
+                System.out.println("---没地方请示了,按不同意处理---\n");
+            }
+        }
+    }
+
+    public void setNextHandler(Handler nextHandler) {
+        this.nextHandler = nextHandler;
+    }
+
+    protected abstract void response(IWomen women);
+
+}
\ No newline at end of file
Index: src/main/java/com/maple/study/pattern/chain/IWomen.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/IWomen.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/IWomen.java	(revision )
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public interface IWomen {
+
+    int getType();
+
+    String getRequest();
+}
Index: src/main/java/com/maple/study/pattern/chain/Father.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/Father.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/Father.java	(revision )
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public class Father extends Handler {
+    public Father() {
+        super(Handler.FATHER_LEVEL_REQUEST);
+    }
+
+    @Override
+    protected void response(IWomen women) {
+        System.out.println("---女儿向父亲请示---");
+        System.out.println(women.getRequest());
+        System.out.println("父亲的答复是: 同意\n");
+    }
+}
Index: src/main/java/com/maple/study/pattern/chain/Son.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/Son.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/Son.java	(revision )
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public class Son extends Handler {
+    public Son() {
+        super(Handler.SON_LEVEL_REQUEST);
+    }
+
+    @Override
+    protected void response(IWomen women) {
+        System.out.println("---母亲向儿子请示---");
+        System.out.println(women.getRequest());
+        System.out.println("儿子的答复是: 同意\n");
+    }
+}
Index: src/main/java/com/maple/study/pattern/chain/Husband.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/maple/study/pattern/chain/Husband.java	(revision )
+++ src/main/java/com/maple/study/pattern/chain/Husband.java	(revision )
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2015. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+ * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
+ * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
+ * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
+ * Vestibulum commodo. Ut rhoncus gravida arcu.
+ */
+
+package com.maple.study.pattern.chain;
+
+/**
+ * Created by aurogon-mac on 15/7/22.
+ */
+public class Husband extends Handler {
+    public Husband() {
+        super(Handler.HUSBAND_LEVEL_REQUEST);
+    }
+
+    @Override
+    protected void response(IWomen women) {
+        System.out.println("---妻子向丈夫请示---");
+        System.out.println(women.getRequest());
+        System.out.println("丈夫的答复是: 同意\n");
+    }
+}
